### ISSUE TITLE
[@typescript-eslint/no-unused-vars] <issue title>

### ISSUE THREAD

**Issue by chrisdrackett** [typescript-eslint/typescript-eslint#786]
**Repro**

```JSON
{
  "rules": {
        "@typescript-eslint/no-unused-vars": [
      "error",
      { "vars": "all", "args": "after-used", "ignoreRestSiblings": false }
    ],
  }
}
```

```TS
import { Dev } from 'components'
```

**Expected Result**

`1:10  error  'Dev' is defined but never used  @typescript-eslint/no-unused-vars`

**Actual Result**

`1:10  warning  'Dev' is defined but never used  @typescript-eslint/no-unused-vars`

**Additional Info**

I can't seem to get this rule to throw an error, only a warning. 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.5` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `12.6.0` |
| `npm`                              | `6.9.0` |

**Comment by chrisdrackett** [comment]
oddly enough it seems that whatever I put for either `no-unused-vars` and `@typescript-eslint/no-unused-vars` is ignored. I've tried:

```
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "off",
```

or even

```
    "no-unused-vars": "error",
    "@typescript-eslint/no-unused-vars": "error",
```

and no matter what I *always* get:

```
1:10 warning 'Dev' is defined but never used @typescript-eslint/no-unused-vars
```

as far as I can tell my other rules work as expected, its only these `no-unused-vars` ones that don't seem to have any effect.

**Comment by chrisdrackett** [comment]
ok, seems this is related to extending `react-app` and the fact that you can't override overrides in pre-6.0.
